azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/Job.java

```diff
@@ -93,33 +98,217 @@
+            /**
+             * Specifies the days of the month when the job will run.
+             * <p>
+             * Use -1 for the last day of a month.
+             *
+             * @param days the job will run an the days of the month specified
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule onTheseDaysOfMonth(int... days);
+
+            /**
+             * Specifies the week days of the month when the job will run.
+             *
+             * @param weekDays the job will run at each week day of the month specified
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule recurringEvery(JobScheduleMonthlyWeekDay... weekDays);
+
+            /**
+             * Specifies the week days of the month when the job will run.
+             *
+             * @param weekDays the job will run at each week day of the month specified
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule recurringEvery(JobScheduleDay... weekDays);
+        }
+
+        /**
+         * The stage of the job definition for an Azure Scheduler allowing to specify advance options for within 24 hours time.
+         */
+        interface WithAdvanceMinutesAndHoursSchedule extends WithEndTime {
+            /**
+             * Specifies the minutes of the hour at which the job will run.
+             *
+             * @param minutes the job will run at each minute specified
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule atTheseMinutes(int... minutes);
+
+            /**
+             * Specifies the hours of the day at which the job will run.
+             * <p>
+             * Hours are specified in 24 hour time.
+             *
+             * @param hours the job will run at each hour (and minute) specified
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule atTheseHours(int... hours);
+        }
+
+        /**
+         * The stage of the job definition for an Azure Scheduler allowing to specify when the job scheduling ends.
+         */
+        interface WithEndTime extends WithAction {
+            /**
+             * Specifies the job to run indefinitely.
+             *
+             * @return the next stage of the definition
+             */
+            WithAction endingNever();
+
+            /**
+             * Specifies the job to complete at a given time.
+             *
+             * @param endTime number of times the job will run
+             * @return the next stage of the definition
+             */
+            WithAction endingBy(DateTime endTime);
+
+            /**
+             * Specifies the job to complete after the specified number of runs.
```

Comment:
similarly:
"Specifies the number of occurrences after which the job will no longer be re-started."